lib/state_machine/state_machine.cpp

```diff
@@ -35,4 +42,43 @@ core::Result StateMachine::handleTransition(const State &state)
   return core::Result::kFailure;
 }
 
+core::Result StateMachine::update()
+{
+  mqtt_->consume();
+  const auto nextMessage = mqtt_->getMessage();
+  if (!nextMessage) { return core::Result::kFailure; }
+
+  const auto doc           = nextMessage->payload;
+  const auto message_state = string_to_state_.find((*doc)["transition"].GetString());
+
+  if (message_state != string_to_state_.end()) {
+    StateMachine::handleTransition(message_state->second);
+    return core::Result::kSuccess;
+  }
+
+  return core::Result::kFailure;
+}
+
+void StateMachine::publishCurrentState()
+{
+  const auto state_string         = StateMachine::stateToString(StateMachine::getCurrentState());
+  const auto topic                = core::MqttTopic::kTest;
+  std::shared_ptr message_payload = std::make_shared<rapidjson::Document>();
+  message_payload->SetObject();
+  rapidjson::Value state_value(state_string.c_str(), message_payload->GetAllocator());
+  message_payload->AddMember("transition", state_value, message_payload->GetAllocator());
+  core::MqttMessage::Header header{.timestamp = 0, .priority = core::MqttMessagePriority::kNormal};
+  const core::MqttMessage message{topic, header, message_payload};
+  mqtt_->publish(message, core::MqttMessageQos::kExactlyOnce);
+}
+
+void StateMachine::startStateMachine()
+{
+  while (StateMachine::getCurrentState() != State::kShutdown) {
+    auto updated = StateMachine::update();
```

Comment:
??

if the return value matters we should do something with it, if not get rid of the return